cmd/neofs-cli/modules/object/range.go

```diff
@@ -161,12 +161,21 @@ func getRangeList(cmd *cobra.Command) ([]*object.Range, error) {
 
 		offset, err := strconv.ParseUint(r[0], 10, 64)
 		if err != nil {
-			return nil, fmt.Errorf("invalid range specifier: %s", vs[i])
+			return nil, fmt.Errorf("invalid '%s' range offset specifier: %w", vs[i], err)
 		}
 		length, err := strconv.ParseUint(r[1], 10, 64)
 		if err != nil {
-			return nil, fmt.Errorf("invalid range specifier: %s", vs[i])
+			return nil, fmt.Errorf("invalid '%s' range length specifier: %w", vs[i], err)
+		}
+
+		if length == 0 {
+			return nil, fmt.Errorf("invalid '%s' range: zero length", vs[i])
 		}
+
+		if offset+length <= offset {
+			return nil, fmt.Errorf("invalid '%s' range: uint64 overflow", vs[i])
```

Comment:
Could you also check the test behaviour? They may check for `out of range` error.

Reply:
seems like tests [want](https://github.com/nspcc-dev/neofs-testcases/blob/d9e881001e2e2f82366851ca26a3c0a448b3b64c/pytest_tests/testsuites/object/test_object_api.py#L417-L425) `OUT_OF_RANGE` here /cc @abereziny

changed to API status return here

Reply:
It's okay to change. We may update testcases to reflect this new error